### PR TITLE
Add db

### PR DESCRIPTION
**Pull request by legidium**

### CODE REVIEW

**Comment by legidium** [comment]
TODO: finish task 4

**Comment by legidium** [comment]
**API Links:**
Get all products: https://2sakgm6pm7.execute-api.eu-west-1.amazonaws.com/dev/products
Get product: https://2sakgm6pm7.execute-api.eu-west-1.amazonaws.com/dev/products/1d9e3589-365a-44e9-aa4e-57a042cd5757

**Comment by legidium** [comment]
**API Links:**
  GET - https://2sakgm6pm7.execute-api.eu-west-1.amazonaws.com/dev/products
  POST - https://2sakgm6pm7.execute-api.eu-west-1.amazonaws.com/dev/products
  GET - https://2sakgm6pm7.execute-api.eu-west-1.amazonaws.com/dev/products/1d9e3589-365a-44e9-aa4e-57a042cd5757

Link to the App: https://d26i3yttsj7uok.cloudfront.net/

**Comment by legidium** [comment]
**API Links:**
  GET - https://2sakgm6pm7.execute-api.eu-west-1.amazonaws.com/dev/products
  POST - https://2sakgm6pm7.execute-api.eu-west-1.amazonaws.com/dev/products
  GET - https://2sakgm6pm7.execute-api.eu-west-1.amazonaws.com/dev/products/1d9e3589-365a-44e9-aa4e-57a042cd5757

Link to the App: https://d26i3yttsj7uok.cloudfront.net/

File: product-service/data/repositories/products-repository-impl.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import {Client, QueryResult} from "pg";
+import {Product, ProductsRepository} from "../../interfaces";
+
+export class ProductsRepositoryImpl implements ProductsRepository {
+  constructor(private db: Client) {}
+
+  async findAll(): Promise<Product[]> {
+    const result: QueryResult<Product> = await this.db.query(`
+      SELECT p.*, s.count FROM products AS p
```
Comment (RidgeA) [inline product-service/data/repositories/products-repository-impl.ts:9]: Just a note - it is not recommended to use asterisks in SQL select queries (except for ad-hoc queries in DB console, of course) 
see - https://stackoverflow.com/questions/15250812/using-asterisk-in-select-queries

Comment (RidgeA) [inline product-service/data/repositories/products-repository-impl.ts:9]: Just a note - it is not recommended to use asterisks in SQL select queries (except for ad-hoc queries in DB console, of course)
see - https://stackoverflow.com/questions/15250812/using-asterisk-in-select-queries

File: product-service/serverless.yml [hunk 2]
```diff
@@ -13,6 +13,12 @@ provider:
 
   environment:
     AWS_NODEJS_CONNECTION_REUSE_ENABLED: 1
+    PG_HOST: lesson-4.cxhfwgzvdjjg.eu-west-1.rds.amazonaws.com
+    PG_PORT: 5432
+    PG_DATABASE: products_service
+    PG_USERNAME: postgres
+    PG_PASSWORD: qwerty123
```
Comment (RidgeA) [inline product-service/serverless.yml:20]: Just a note - it is not safe to publish DB password to VCS

**Review by RidgeA — APPROVED** [review]
LGTM

File: product-service/data/repositories/products-repository-impl.ts (hunk 1 above)
Comment (legidium) [reply product-service/data/repositories/products-repository-impl.ts:9]: Sure, I know about asterisks and never use them in real production apps, only as a shortcut in demo apps.
Anyway, thanks for your note.

File: product-service/serverless.yml (hunk 2 above)
Comment (legidium) [reply product-service/serverless.yml:20]: My bad, it was a late night commit. But qwerty123 is not a real password )
Nevertheless your're right, I should'nt have put it in the commit.

**Pull request merged**